diff --git a/tables/CMakeLists.txt b/tables/CMakeLists.txt
index 495f63f..c76d84c 100644
--- a/tables/CMakeLists.txt
+++ b/tables/CMakeLists.txt
@@ -106,18 +106,18 @@ MACRO(CONVERT_DB targetName)
 	DESTINATION ${ICON_INSTALL_DIRECTORY})
 ENDMACRO(CONVERT_DB targetName)
 
-ADD_SUBDIRECTORY(array)
+#ADD_SUBDIRECTORY(array)
 ADD_SUBDIRECTORY(cangjie)
-ADD_SUBDIRECTORY(cantonese)
-ADD_SUBDIRECTORY(easy)
-ADD_SUBDIRECTORY(erbi)
+#ADD_SUBDIRECTORY(cantonese)
+#ADD_SUBDIRECTORY(easy)
+#ADD_SUBDIRECTORY(erbi)
 ADD_SUBDIRECTORY(quick)
-ADD_SUBDIRECTORY(scj)
-ADD_SUBDIRECTORY(stroke5)
-ADD_SUBDIRECTORY(wu)
-ADD_SUBDIRECTORY(wubi-haifeng)
-ADD_SUBDIRECTORY(wubi-jidian)
-ADD_SUBDIRECTORY(yong)
+#ADD_SUBDIRECTORY(scj)
+#ADD_SUBDIRECTORY(stroke5)
+#ADD_SUBDIRECTORY(wu)
+#ADD_SUBDIRECTORY(wubi-haifeng)
+#ADD_SUBDIRECTORY(wubi-jidian)
+#ADD_SUBDIRECTORY(yong)
 
 #CONVERT_DB(yong "yong/yong.txt")
 
